yt/geometry/coordinates/cartesian_coordinates.py

```diff
@@ -21,13 +21,37 @@
     _get_vert_fields, \
     cartesian_to_cylindrical, \
     cylindrical_to_cartesian
+from yt import YTArray
 from yt.funcs import mylog
+from yt.units.yt_array import uvstack
 from yt.utilities.lib.pixelization_routines import \
     pixelize_element_mesh, pixelize_off_axis_cartesian, \
-    pixelize_cartesian, pixelize_cartesian_nodal
+    pixelize_cartesian, pixelize_cartesian_nodal, \
+    element_mesh_line_plot
 from yt.data_objects.unstructured_mesh import SemiStructuredMesh
 from yt.utilities.nodal_data_utils import get_nodal_data
 
+def _sample_ray(ray, resolution, field):
+    start_point = ray.start_point
+    end_point = ray.end_point
+    sample_dr = (end_point - start_point)/(resolution-1)
+    sample_points = [np.arange(resolution)*sample_dr[i] for i in range(3)]
+    sample_points = uvstack(sample_points).T
+    ray_coordinates = uvstack([ray[d] for d in 'xyz']).T
+    ray_dds = uvstack([ray['d'+d] for d in 'xyz']).T
+    ray_field = ray[field]
+    field_values = ray.ds.arr(np.zeros(resolution), ray_field.units)
+    for i, sample_point in enumerate(sample_points):
+        ray_contains = ((sample_point >= (ray_coordinates - ray_dds/2)) &
+                        (sample_point <= (ray_coordinates + ray_dds/2)))
+        ray_contains = ray_contains.all(axis=-1)
+        wh = np.where(ray_contains)[0]
```

Comment:
Is this `np.where` operation necessary?  Could you index `ray_field` with the `ray_contains` boolean array?  It would add a small bit of speed by saving an operation.